### DISCUSSION TITLE
group by with  datefield function bug

### DISCUSSION THREAD

**Discussion by jeffdico** [sqlalchemy/sqlalchemy#11439] [Usage Questions]
### Describe the bug

I wrote a query and grouped it by a datefield with date_part or extract or to_char  and group by the datefield with the respective function. But the library was expecting a group by of the full date field instead and was throwing an error. Meanwhile this exact query written in sql  text and passed worked very well. 

here is the query code. 

```
      qry = select(
            func.to_char(SampleTable.date_enrolled, 'mm/dd')
        ).where(
            SampleTable.date_enrolled.between(start, end)
        ).group_by(func.to_char(SampleTable.date_enrolled, 'mm/dd'))
        
        out = session.execute(qry) # error is triggered at this point. 

        # if this query is changed to a text sql, it will work very well no issues
```

I have written similar query in the past and it worked very well. I bet this will work for normal pyscopg2 driver. This error can be recreated using the **asyncpg** driver.  

### SQLAlchemy Version in Use

2.0.25

### DBAPI (i.e. the database driver)

asyncpg

### Database Vendor and Major Version

postgresql

### Python Version

3.11

### Operating system

Linux

### To Reproduce

```python
qry = select(
            func.to_char(SampleTable.date_enrolled, 'mm/dd')
        ).where(
            SampleTable.date_enrolled.between(start, end)
        ).group_by(func.to_char(SampleTable.date_enrolled, 'mm/dd'))
        
  out = session.execute(qry) # error is triggered at this point.
```

### Error

```
# Copy the complete stack trace and error message here, including SQL log output if applicable.
```

**Comment by CaselIT** [comment]
Hi,

The query uses bound parameters, so you have to either re-use the sqlalchemy same function in both places (so that they share the parameter) or use the label in the group by https://docs.sqlalchemy.org/en/20/tutorial/data_select.html#tutorial-order-by-label
